### PR TITLE
forked from master, added dsymm, stats_correlation/covariance tests

### PR DESCRIPTION
**Pull request by lijun99**
Start a new branch gsl directly from master, and add the test routines. 
(I will delete gsl_stats branch).

### CODE REVIEW

File: tests/gsl/stats_correlation.py [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    # set them to random values
+    rng = gsl.rng()
+    v1.random(pdf=gsl.pdf.uniform(support=(-1,1), rng=rng))
+    v2.random(pdf=gsl.pdf.uniform(support=(-1,1), rng=rng))
+
+    # call correlation
+    correlation = gsl.stats.correlation(v1, v2)
+
+    # correlation with itself should be one
+    # assert gsl.stats.correlation(v1, v1) == 1.0
```
Comment (aivazis) [inline tests/gsl/stats_correlation.py]: why is this commented out?

File: tests/gsl/blas_dsymm.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    # the matrix C
+    C = gsl.matrix(shape=(2,3))
+    C[0,0], C[0,1], C[0,2] = 0,0,0
+    C[1,0], C[1,1], C[1,2] = 0,0,0
+
+    # compute the form
+    gsl.blas.dsymm(A.sideLeft, A.upperTriangular, α, A, B, β, C)
+
+    # check
+    # print(tuple(y))
```
Comment (aivazis) [inline tests/gsl/blas_dsymm.py]: can you remove this line? there doesn't seem to be a γ in this routine

File: tests/gsl/stats_covariance.py [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    v2 = gsl.vector(shape=length)
+    # set them to random values
+    rng = gsl.rng()
+    v1.random(pdf=gsl.pdf.uniform(support=(-1,1), rng=rng))
+    v2.random(pdf=gsl.pdf.uniform(support=(-1,1), rng=rng))
+
+    # call correlation
+    covariance = gsl.stats.covariance(v1, v2)
+
+    # assert gsl.stats.covariance(v1, v1) == v1.variance(mean=v1.mean())
```
Comment (aivazis) [inline tests/gsl/stats_covariance.py]: this assertion is also commented out; why?

**Comment by lijun99** [comment]
Changed the code accordingly.

**Pull request merged**